doc/devel/coding_style_guideline.rst

```diff
@@ -0,0 +1,169 @@
+  dictionary. Make exceptions for ``id`` and documented domain-specific
+  abbreviations (e.g. ``fa``, ``md`, ``acc``, ``pos``).
+* Names are usually read left-to-right, from more specific to less specific:
+  e.g. ``L2RegularizedSphericalDeconvolutionFilter`` instead of
+  ``SphericalDeconvolutionL2RegularizedFilter``.
+
+Package names
+^^^^^^^^^^^^^
+
+* Use a noun name: e.g. ``learning`` instead of ``learners``. An exception could
+  be a package named ``models``.
+
+Class names
+^^^^^^^^^^^
+
+* Use a noun-phrase name: e.g. ``GaussianNoiseRemovalFilter`` instead of
+  ``GaussianNoiseFiltering``.
+
+Method names
+^^^^^^^^^^^^
+
+* Use a verb-phrase for method names: e.g. ``flip_streamlines`` (instead of
+  ``streamline_flip``).
+* Use the ``get`` prefix for field accessors that return a value.
+* Use ``is`` and ``has`` prefixes for boolean field accessors.
+* Use the ``set`` prefix for field accessors that do not return a value.
+* Use validation verbs for methods that provide the result: use verbs like
+  ``validate``, ``check`` or ``ensure`` to name methods that either result or
+  throw an exception when validation fails.
+* Use transformation verbs for methods that return a transformed value: use
+  verbs that suggest transformation, like ``convert``, for methods that return
+  the result.
+* Use the ``compute`` prefix for methods that compute and return a value.
+* Use opposites precisely, e.g. ``add/remove``, ``begin/end``, ``first/last``,
+  ``insert/delete``, etc.
+* Use the name ``create`` instead of ``generate`` for methods that *create*
+  something.
+
+Variables
+^^^^^^^^^
+
+* Qualification: qualify values with suffixes. Use a suffix to describe what
+  kind of value constant and variable values represent. Suffixes such as
+  ``min`` (for maximum) or ``min`` (for minimum), ``count`` (instead of ``num``
+  or ``nb``) and ``avg`` (for average) relate a collection of values to a single
+  derived value. Using a suffix, rather than a prefix, for the qualifier
+  naturally links the name to other similar names: e.g.
+  ``streamline_lengths_max`` instead of ``max_streamline_lengths``, or
+  ``streamline_count`` instead of ``count_streamlines`` (which would be a method
+  name).
+* Avoid adding the type of a variable to its name, i.e. do not use::
+
+    streamlines_list = []
+
+  Instead, use::
+
+    streamlines = []
+
+* Limit the use of plural words to collections, but prefer collective nouns for
+  collections whenever possible. If a variable comprising two names needs to b
+  pluralized, pluralize the last term, i.e. ``streamline_lengths`` instead of
+  ``streamlines_length``.
+* Use boolean variable names that imply true or false: use names like ``done``
+  or ``found`` that describe boolean values.
+* Replace boolean names with names in the correct grammatical form: do not use
+  negation in boolean names. Do not use names that require a prefix like ``not``
+  that inverts the variable’s truth value.
+
+Documentation
```

Comment:
Maybe I would move this in a separated page? python coding styyle and rst coding style.

Reply:
For now, I would avoid splitting it, since that would mean that we should have three files, the parent including the two childs (code; rst), and I think it is fine as it is at least for now. Also, the documentation is intimately linked to the code.